pytest/tests/sanity/split_storage.py

```diff
@@ -76,6 +76,126 @@ def _check_split_storage_info(
         else:
             self.assertIsNone(cold_head_height)
 
+    def _migrate_to_split_storage(self, rpc, archival, archival_dir,
+                                  wait_period):
+
+        # Wait until a few blocks are produced so that we're sure that the db is
+        # properly created and populated with some data.
+        n = 5
+        n = wait_for_blocks(archival, target=n).height
+
+        self._check_split_storage_info(
+            "migration_phase_1",
+            node=archival,
+            expected_head_height=n,
+            # The hot db kind should remain archive until fully migrated.
+            expected_hot_db_kind="Archive",
+            # The cold storage is not configured so cold head should be none.
+            check_cold_head=False,
+        )
+
+        logger.info("")
+        logger.info("Phase 2 - Setting the cold storage and restarting neard.")
+        logger.info("")
+
+        self._configure_cold_storage(archival_dir)
+        archival.kill(gentle=True)
+        archival.start()
+
+        # Wait for a few seconds to:
+        # - Give the node enough time to get started.
+        # - Give the cold store loop enough time to run - it runs every 1s.
+        # TODO(wacban) this is a quick stop-gap solution to fix nayduck, this
+        # should be solved by waiting in a loop until cold store head is at
+        # expected proximity to final head.
+        time.sleep(4)
+
+        # Wait until enough blocks so that we produce enough blocks to fill 5
+        # epochs to trigger GC - otherwise the tail won't be set.
+        n = max(n, wait_period) + 5
+        logger.info(f"Wait until RPC reaches #{n}")
+        wait_for_blocks(rpc, target=n)
+        logger.info(f"Wait until archival reaches #{n}")
+        wait_for_blocks(archival, target=n)
+
+        self._check_split_storage_info(
+            "migration_phase_2",
+            node=archival,
+            expected_head_height=n,
+            # The hot db kind should remain archive until fully migrated.
+            expected_hot_db_kind="Archive",
+            # The cold storage head should be fully caught up by now.
+            check_cold_head=True,
+        )
+
+        logger.info("")
+        logger.info("Phase 3 - Preparing hot storage from rpc backup.")
+        logger.info("")
+
+        # TODO Ideally we won't need to stop the node while running prepare-hot.
```

Comment:
How can you not stop the node to do the following manipulations?
You're making a massive write to the database.

Reply:
prepare-hot command only changes the DBKind of provided rpc db. Archival node is needed to make sure that rpc db is compatible with it as future hot db.
So, actually archival node can keep running. And actually it already can (this code was written before some changes in cold store tool). Still, archival node will need a restart in the next phase, so the kill command just moves here.